pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,75 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/machine/libmachine/drivers"
+)
+
+var (
+	pullSecret = `apiVersion: v1
+data:
+  .dockerconfigjson: %s
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: openshift-config
+type: kubernetes.io/dockerconfigjson`
+
+	// This command timeout in 80 secs if not able to replace the pull
+	// secret. We first start the kubelet and wait till the api server is
+	// up and running by hitting it with replace command.
+	replacePullSecretCmd = `timeout 80 bash -c 'sudo systemctl start kubelet && \
+until oc --config /tmp/kubeconfig replace -f /tmp/pull-secret.yaml 2>/dev/null 1>&2; \
+do echo "Waiting for recovery apiserver to come up."; sleep 1; done && \
+sudo systemctl stop kubelet && \
+sudo crictl stopp $(sudo crictl pods -q) && \
+sudo crictl rmp $(sudo crictl pods -q)'`
+)
+
+func AddPullSecret(driver drivers.Driver, pullSec string, kubeconfig string) error {
+	if err := addKubeconfigFileToInstance(driver, kubeconfig); err != nil {
+		return err
+	}
+	if err := addpullSecretFileToInstance(driver, pullSec); err != nil {
+		return err
+	}
+	output, err := drivers.RunSSHCommandFromDriver(driver, replacePullSecretCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", replacePullSecretCmd, output)
+
+	if err := addpullSecretToIntanceDisk(driver, pullSec); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addpullSecretFileToInstance(driver drivers.Driver, pullSec string) error {
+	base64OfPullSec := base64.StdEncoding.EncodeToString([]byte(pullSec))
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/pull-secret.yaml\n%s\nEOF", fmt.Sprintf(pullSecret, base64OfPullSec)))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addpullSecretToIntanceDisk(driver drivers.Driver, pullSec string) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | sudo tee /var/lib/kubelet/config.json\n%s\nEOF", pullSec))
```

Comment:
The `oc replace` call is not triggering the recreation of this file?

Reply:
@cfergeau `oc replace` call is only replace the kubernetes resource, it is not going to affect this file since this is on disk and not mounted with any kubenetes resources.

Reply:
Yes, I was hoping that the process 'owning' this file would notice the kubernetes resource change, and rewrite it. Do you know which process is creating it on a fresh openshift install?